plugins/osdn/common.go

```diff
@@ -95,105 +30,6 @@ func isMultitenantPlugin(pluginType string) bool {
 	}
 }
 
-func (oc *OsdnNode) Start(mtu uint) error {
-	// Assume we are working with IPv4
-	clusterNetwork, err := oc.Registry.GetClusterNetwork()
-	if err != nil {
-		log.Errorf("Failed to obtain ClusterNetwork: %v", err)
-		return err
-	}
-
-	ipt := iptables.New(kexec.New(), utildbus.New(), iptables.ProtocolIpv4)
-	if err := SetupIptables(ipt, clusterNetwork.String()); err != nil {
-		return fmt.Errorf("Failed to set up iptables: %v", err)
-	}
-
-	ipt.AddReloadFunc(func() {
-		err := SetupIptables(ipt, clusterNetwork.String())
-		if err != nil {
-			log.Errorf("Error reloading iptables: %v\n", err)
-		}
-	})
-
-	networkChanged, err := oc.SubnetStartNode(mtu)
-	if err != nil {
-		return err
-	}
-
-	if oc.multitenant {
-		if err := oc.VnidStartNode(); err != nil {
-			return err
-		}
-	}
-
-	if networkChanged {
-		pods, err := oc.GetLocalPods(kapi.NamespaceAll)
-		if err != nil {
-			return err
-		}
-		for _, p := range pods {
-			containerID := GetPodContainerID(&p)
-			err = oc.UpdatePod(p.Namespace, p.Name, kubeletTypes.DockerID(containerID))
-			if err != nil {
-				log.Warningf("Could not update pod %q (%s): %s", p.Name, containerID, err)
-			}
-		}
-	}
-
-	oc.markPodNetworkReady()
-
-	return nil
-}
-
-func (oc *OsdnNode) GetLocalPods(namespace string) ([]kapi.Pod, error) {
-	return oc.Registry.GetRunningPods(oc.HostName, namespace)
-}
-
-func (oc *OsdnNode) markPodNetworkReady() {
-	close(oc.podNetworkReady)
-}
-
-func (oc *OsdnNode) WaitForPodNetworkReady() error {
-	logInterval := 10 * time.Second
-	numIntervals := 12 // timeout: 2 mins
-
-	for i := 0; i < numIntervals; i++ {
-		select {
-		// Wait for StartNode() to finish SDN setup
-		case <-oc.podNetworkReady:
-			return nil
-		case <-time.After(logInterval):
-			log.Infof("Waiting for SDN pod network to be ready...")
-		}
-	}
-	return fmt.Errorf("SDN pod network is not ready(timeout: 2 mins)")
-}
-
-type FirewallRule struct {
-	table string
-	chain string
-	args  []string
-}
-
-func SetupIptables(ipt iptables.Interface, clusterNetworkCIDR string) error {
-	rules := []FirewallRule{
-		{"nat", "POSTROUTING", []string{"-s", clusterNetworkCIDR, "!", "-d", clusterNetworkCIDR, "-j", "MASQUERADE"}},
-		{"filter", "INPUT", []string{"-p", "udp", "-m", "multiport", "--dports", "4789", "-m", "comment", "--comment", "001 vxlan incoming", "-j", "ACCEPT"}},
-		{"filter", "INPUT", []string{"-i", "tun0", "-m", "comment", "--comment", "traffic from docker for internet", "-j", "ACCEPT"}},
-		{"filter", "FORWARD", []string{"-d", clusterNetworkCIDR, "-j", "ACCEPT"}},
-		{"filter", "FORWARD", []string{"-s", clusterNetworkCIDR, "-j", "ACCEPT"}},
-	}
-
-	for _, rule := range rules {
-		_, err := ipt.EnsureRule(iptables.Prepend, iptables.Table(rule.table), iptables.Chain(rule.chain), rule.args...)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
 func GetNodeIP(node *kapi.Node) (string, error) {
```

Comment:
We could move GetNodeIP() to plugins/osdn/subnets.go, that's where we handle node related stuff.